HLTrigger/JetMET/plugins/HLTScoutingPFProducer.cc

```diff
@@ -155,36 +160,71 @@ void HLTScoutingPFProducer::produce(edm::StreamID sid, edm::Event &iEvent, edm::
             break;
           ++index_counter;
         }
+        float normchi2{0}, dz{0}, dxy{0}, dzError{0}, dxyError{0}, trk_pt{0}, trk_eta{0}, trk_phi{0};
+        uint8_t lostInnerHits{0}, quality{0};
+        if (doTrackRelVars_) {
+          const auto *trk = cand.bestTrack();
+          if (trk != nullptr) {
+            normchi2 = MiniFloatConverter::reduceMantissaToNbitsRounding(std::floor(trk->normalizedChi2()),
+                                                                         mantissaPrecision_);
+            lostInnerHits = btagbtvdeep::lost_inner_hits_from_pfcand(cand);
+            quality = btagbtvdeep::quality_from_pfcand(cand);
+            trk_pt = MiniFloatConverter::reduceMantissaToNbitsRounding(trk->pt(), mantissaPrecision_);
+            trk_eta = MiniFloatConverter::reduceMantissaToNbitsRounding(trk->eta(), mantissaPrecision_);
+            trk_phi = MiniFloatConverter::reduceMantissaToNbitsRounding(trk->phi(), mantissaPrecision_);
+
+            if (not vertexCollection->empty()) {
+              const reco::Vertex &pv = (*vertexCollection)[0];
 
-        outPFCandidates->emplace_back(MiniFloatConverter::reduceMantissaToNbitsRounding(cand.pt(), mantissaPrecision),
-                                      MiniFloatConverter::reduceMantissaToNbitsRounding(cand.eta(), mantissaPrecision),
-                                      MiniFloatConverter::reduceMantissaToNbitsRounding(cand.phi(), mantissaPrecision),
-                                      MiniFloatConverter::reduceMantissaToNbitsRounding(cand.mass(), mantissaPrecision),
-                                      cand.pdgId(),
-                                      vertex_index);
+              dz = MiniFloatConverter::reduceMantissaToNbitsRounding(trk->dz(pv.position()), mantissaPrecision_);
+              dxy = MiniFloatConverter::reduceMantissaToNbitsRounding(trk->dxy(pv.position()), mantissaPrecision_);
+              dzError = MiniFloatConverter::reduceMantissaToNbitsRounding(dz / trk->dzError(), mantissaPrecision_);
+              dxyError = MiniFloatConverter::reduceMantissaToNbitsRounding(dxy / trk->dxyError(), mantissaPrecision_);
```

Comment:
```suggestion
              dz = trk->dz(pv.position());
              dzError = MiniFloatConverter::reduceMantissaToNbitsRounding(dz / trk->dzError(), mantissaPrecision_);
              dz = MiniFloatConverter::reduceMantissaToNbitsRounding(dz, mantissaPrecision_);

              dxy = trk->dxy(pv.position());
              dxyError = MiniFloatConverter::reduceMantissaToNbitsRounding(dxy / trk->dxyError(), mantissaPrecision_);
              dxy = MiniFloatConverter::reduceMantissaToNbitsRounding(dxy, mantissaPrecision_);
```
In the current impl, `dzError` is calculated based on a `dz` with reduced precision (same goes for `dxy`). In principle, that's not needed, but I didn't check if this makes a significant difference.

Reply:
I just did a quick check and it seem to have at least a small effect. Better to be safe, I'll commit it.